### PR TITLE
[REVIEW] Update cuda-python dependency to 11.7.1

### PR DESCRIPTION
**Pull request by galipremsagar**
## Description
This is a mirror PR of https://github.com/rapidsai/cudf/pull/11994 to unblock gpu-ci which is currently blocked.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: ci/cpu/build.sh [hunk 1]
```diff
@@ -84,6 +86,9 @@ else
   CONDA_CHANNEL="-c $WORKSPACE/ci/artifacts/cudf/cpu/.conda-bld/"
 fi
 
+# TODO: remove
+sed -i '/rapidsai$/d' /opt/conda/.condarc
+
```
Comment (galipremsagar) [inline ci/cpu/build.sh:91]:
```suggestion
# TODO: remove
#sed -i '/rapidsai$/d' /opt/conda/.condarc

```

File: conda/recipes/cudf/meta.yaml [hunk 2]
```diff
@@ -41,8 +41,9 @@ requirements:
     - numba >=0.56.2
     - dlpack>=0.5,<0.6.0a0
     - pyarrow =9
+    - protobuf>=3.20.1,<3.21.0a0
     - libcudf ={{ version }}
-    - rmm ={{ minor_version }}
+    - rapidsai-nightly::rmm ={{ minor_version }}
```
Comment (galipremsagar) [inline conda/recipes/cudf/meta.yaml:46]:
```suggestion
    - rmm ={{ minor_version }}
```

File: conda/recipes/libcudf/meta.yaml [hunk 3]
```diff
@@ -32,10 +32,12 @@ requirements:
     - {{ compiler('cxx') }}
     - {{ compiler('cuda') }} {{ cuda_version }}
     - sysroot_{{ target_platform }} {{ sysroot_version }}
+    - libprotobuf>=3.20.1,<3.21.0a0
   host:
-    - librmm {{ minor_version }}.*
+    - rapidsai-nightly::librmm {{ minor_version }}.*
```
Comment (galipremsagar) [inline conda/recipes/libcudf/meta.yaml:37]:
```suggestion
    - librmm {{ minor_version }}.*
```

File: conda/recipes/libcudf/meta.yaml [hunk 4]
```diff
@@ -55,8 +57,9 @@ outputs:
         - cmake {{ cmake_version }}
       run:
         - cudatoolkit {{ cuda_spec }}
-        - librmm {{ minor_version }}.*
+        - rapidsai-nightly::librmm {{ minor_version }}.*
```
Comment (galipremsagar) [inline conda/recipes/libcudf/meta.yaml:60]:
```suggestion
        - librmm {{ minor_version }}.*
```

**Comment by galipremsagar** [comment]
rerun tests

File: conda/recipes/libcudf/meta.yaml [hunk 5]
```diff
@@ -32,10 +32,12 @@ requirements:
     - {{ compiler('cxx') }}
     - {{ compiler('cuda') }} {{ cuda_version }}
     - sysroot_{{ target_platform }} {{ sysroot_version }}
+    - libprotobuf>=3.20.1,<3.21.0a0
   host:
     - librmm {{ minor_version }}.*
     - cudatoolkit {{ cuda_version }}.*
     - arrow-cpp {{ arrow_cpp_version }}
+    - libprotobuf>=3.20.1,<3.21.0a0
```
Comment (galipremsagar) [inline conda/recipes/libcudf/meta.yaml:40]:
```suggestion
    #- libprotobuf>=3.20.1,<3.21.0a0
```

File: ci/cpu/build.sh [hunk 6]
```diff
@@ -49,6 +49,8 @@ gpuci_logger "Activate conda env"
 . /opt/conda/etc/profile.d/conda.sh
 conda activate rapids
 
+gpuci_mamba_retry install rapidsai-nightly::rmm==22.10.00a221026 rapidsai-nightly::librmm==22.10.00a221026
+
```
Comment (galipremsagar) [inline ci/cpu/build.sh:53]:
```suggestion
#gpuci_mamba_retry install rapidsai-nightly::rmm==22.10.00a221026 rapidsai-nightly::librmm==22.10.00a221026

```

**Comment by galipremsagar** [comment]
rerun tests

**Review by ajschmidt8 — APPROVED** [review]

**Comment by galipremsagar** [comment]
@gpucibot merge

**Pull request merged**